File: src/main/java/no/nav/arbeidsgiver/sykefravarsstatistikk/api/statistikk/sykefraværshistorikk/aggregert/Trend.java
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.statistikk.sykefraværshistorikk.aggregert;

import java.math.BigDecimal;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Data;
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.felles.ÅrstallOgKvartal;
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.statistikk.Statistikkategori;

@Data
@AllArgsConstructor
public class Trend {

    public BigDecimal trendverdi;
    public int antallPersonerIBeregningen;
    public List<ÅrstallOgKvartal> kvartalerIBeregningen;

    StatistikkDto tilAggregertHistorikkDto(
            Statistikkategori type, String label) {
        return StatistikkDto.builder()
                .statistikkategori(type)
                .label(label)
                .verdi(this.trendverdi.toString())
                .antallPersonerIBeregningen(this.antallPersonerIBeregningen)
                .kvartalerIBeregningen(this.kvartalerIBeregningen)
                .build();
    }
}

File: src/main/java/no/nav/arbeidsgiver/sykefravarsstatistikk/api/statistikk/sykefraværshistorikk/aggregert/Trend.kt
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.statistikk.sykefraværshistorikk.aggregert

import no.nav.arbeidsgiver.sykefravarsstatistikk.api.felles.ÅrstallOgKvartal
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.statistikk.Statistikkategori
import java.math.BigDecimal

data class Trend(
    var trendverdi: BigDecimal? = null,
    var antallPersonerIBeregningen: Int = 0,
    var kvartalerIBeregningen: List<ÅrstallOgKvartal>? = null
) {
    fun tilAggregertHistorikkDto(type: Statistikkategori?, label: String?): StatistikkDto {
        return StatistikkDto(
            statistikkategori = type,
            label = label,
            verdi = trendverdi.toString(),
            antallPersonerIBeregningen = antallPersonerIBeregningen,
            kvartalerIBeregningen = kvartalerIBeregningen
        )
    }
}